pkg/clients/elasticache/elasticache.go

```diff
@@ -368,3 +375,23 @@ func isErrorCodeEqual(errorCode string, err error) bool {
 
 	return ce.Code() == errorCode
 }
+
+// IsSubnetGroupUpToDate checks if CacheSubnetGroupParameters are in sync with provider values
+func IsSubnetGroupUpToDate(p cachev1alpha1.CacheSubnetGroupParameters, sg elasticache.CacheSubnetGroup) bool {
+	subnetEqual := len(p.SubnetIds) == len(sg.Subnets) && (func() bool {
+		r := false
+		for _, id := range p.SubnetIds {
+			for _, subnet := range sg.Subnets {
+				if id == *subnet.SubnetIdentifier {
+					r = true
+					break
+				} else {
+					r = false
+				}
+			}
+		}
+		return r
+	})()
+
+	return subnetEqual && p.Description == aws.StringValue(sg.CacheSubnetGroupDescription)
```

Comment:
It seems like the goal here is to determine whether the list of subnets is equal, irrespective of order. Performance probably doesn't matter that much given that there's not likely to be more than a handful of subnets, but we could avoid the `O(n^m)` nested loop by turning one of the inputs into a set, for example:

```go
func IsSubnetGroupUpToDate(p cachev1alpha1.CacheSubnetGroupParameters, sg elasticache.CacheSubnetGroup) bool {
	if p.Description != aws.StringValue(sg.CacheSubnetGroupDescription) {
		return false
	}

	if len(p.SubnetIds) != len(sg.Subnets) {
		return false
	}

	exists := make(map[string]bool)
	for _, id := range sg.Subnets {
		exists[id] = true
	}
	for _, id := range p.SubnetIds {
		if !exists[id] {
			return false
		}
	}

	return true
}
```